### PR TITLE
chore: Upgrade Cypress to v9

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

We should do this as it speeds up the tests quite a bit! 🚀 
This also sets us up nicely for Cypress v10: https://docs.cypress.io/guides/references/changelog#10-0-0

List of changes:

- Upgrade Cypress docker image in CircleCI to use newer Chrome
- Upgrade Gatsby deps like `gatsby-cypress` to `next`
- Update test assertions where needed
- Remove unused `gatsby-image` e2e test

Unfortunately I had to remove one test: `production-runtime/prefetching` the `slow` variant. The `--force-effective-connection-type` flag doesn't exist anymore in modern Chrome browsers so we can't use that. Cypress tells you in e.g. https://www.cypress.io/blog/2020/11/12/testing-application-in-offline-network-mode/ that you can use `remote:debugger:protocol` to hook into devtools. Then one could use `Network.emulateNetworkConditions` with `connectionType` (see https://chromedevtools.github.io/devtools-protocol/tot/Network/#method-emulateNetworkConditions).

But that doesn't work in `cypress run` and/or is broken all together (see https://github.com/cypress-io/cypress/issues/17723). Simulating offline mode (which is in the same vein) also doesn't work: https://github.com/cypress-io/cypress/issues/235

[ch52003]

### CODE REVIEW

File: e2e-tests/production-runtime/cypress/integration/compilation-hash.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ describe(
 
         cy.visit(`/`).waitForRouteChange()
         let didMock = false
-        cy.intercept("/app-data.json", req => {
+        cy.intercept("/page-data/app-data.json", req => {
```
Comment (LekoArts) [inline e2e-tests/production-runtime/cypress/integration/compilation-hash.js:63]: https://docs.cypress.io/guides/references/migration-guide#URL-matching-is-stricter

File: e2e-tests/production-runtime/cypress/integration/resource-loading-resilience.js [hunk 2]
```diff
@@ -2,6 +2,12 @@ const waitForAPIOptions = {
   timeout: 5000,
 }
 
+Cypress.on('uncaught:exception', () => {
```
Comment (LekoArts) [inline e2e-tests/production-runtime/cypress/integration/resource-loading-resilience.js:5]: https://docs.cypress.io/guides/references/migration-guide#Uncaught-exception-and-unhandled-rejections

**Comment by j0sh77** [comment]
(haven't gotten to the code yet... but) I don't recall this one being flaky, so perhaps there's a breaking change here?

```
  Running:  gatsby-plugin-image.js                                                         (8 of 38)
  Estimated: 2 seconds


  gatsby-plugin-image
    1) doesn't recycle image nodes when not necessary
    ✓ rerenders when image src changed (770ms)
    ✓ rerenders when background color changes (741ms)


  2 passing (2s)
  1 failing

  1) gatsby-plugin-image
       doesn't recycle image nodes when not necessary:
     AssertionError: expected stub to have been called with arguments [Object{3}], but it was never called
      at Context.eval (http://localhost:9000/__cypress/tests?p=cypress/integration/gatsby-plugin-image.js:201:34)
```

Also quite a few under react 18 tests 🤔 

Looking through the code shortly!

**Comment by LekoArts** [comment]
The React 18 failures seem to be https://reactjs.org/docs/error-decoder.html/?invariant=425 - I have to check this out locally

**Comment by j0sh77** [comment]
So far code change looks good to me! Thanks for doing this. 😄

**Comment by LekoArts** [comment]
Great 🙈 

For some reason Cypress doesn't set the 2G connection type in headless mode

```shell
Prefetching
    1) should not prefetch if on slow connection


  0 passing (11s)
  1 failing

  1) Prefetching
       should not prefetch if on slow connection:

      Timed out retrying after 4000ms
      + expected - actual

      -'4g'
      +'2g'
```

**Review by tyhopp — APPROVED** [review]
LGTM, the current failing tests are existing flakes IIRC so this should be good to go!

**Pull request merged**